internal/fsutil/helpers_test.go

```diff
@@ -0,0 +1,37 @@
+package fsutil_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-controller/internal/fsutil"
+)
+
+func TestEnsureEmptyDirectory(t *testing.T) {
+	tempDir := t.TempDir()
+	dirPath := filepath.Join(tempDir, "testdir")
+
+	// Ensure directory is created if not exists
+	require.NoError(t, fsutil.EnsureEmptyDirectory(dirPath, 0755))
+
+	stat, err := os.Stat(dirPath)
+	require.NoError(t, err)
+	require.True(t, stat.IsDir())
+
+	// Create files inside directory
+	file := filepath.Join(dirPath, "file1")
+	require.NoError(t, os.WriteFile(file, []byte("test"), 0600))
+
+	subDir := filepath.Join(dirPath, "subdir")
+	require.NoError(t, os.Mkdir(subDir, 0755))
+
+	// Ensure directory is emptied but not deleted
+	require.NoError(t, fsutil.EnsureEmptyDirectory(dirPath, 0755))
+
+	entries, err := os.ReadDir(dirPath)
+	require.NoError(t, err)
+	require.Empty(t, entries)
+}
```

Comment:
```suggestion
func TestEnsureEmptyDirectory(t *testing.T) {
	tempDir := t.TempDir()
	dirPath := filepath.Join(tempDir, "testdir")

	t.Log("Ensuring the directory is created if it does not exist")
	require.NoError(t, fsutil.EnsureEmptyDirectory(dirPath, 0755))

	stat, err := os.Stat(dirPath)
	require.NoError(t, err)
	require.True(t, stat.IsDir())

	t.Log("Creating files inside the directory")
	file := filepath.Join(dirPath, "file1")
	require.NoError(t, os.WriteFile(file, []byte("test"), 0600))

	subDir := filepath.Join(dirPath, "subdir")
	require.NoError(t, os.Mkdir(subDir, 0755))

	t.Log("Ensuring the directory is emptied but not deleted")
	require.NoError(t, fsutil.EnsureEmptyDirectory(dirPath, 0755))

	entries, err := os.ReadDir(dirPath)
	require.NoError(t, err)
	require.Empty(t, entries)

	t.Log("Test completed successfully, directory is empty")
}
```
👍 Good catcher. But could we replace the comments with logs so that if fails we have further information and we also clarify the steps?